tests/sentry/testutils/helpers/test_features.py

```diff
@@ -56,3 +56,39 @@ def test_feature_with_rpc_organization(self):
             assert isinstance(org_context.organization, RpcOrganization)
 
             assert features.has("system:multi-region")
+
+            assert features.has("system:multi-region")
+
+
+class TestWithFeatureClassDecorator(TestCase):
+    """Test that with_feature works correctly when used as a class decorator."""
+
+    def test_with_feature_on_class_works(self):
+        """Test that using with_feature as a class decorator enables features for all methods."""
+
+        @with_feature("organizations:global-views")
+        class TestClassWithFeature(TestCase):
+            def test_method_1(self):
+                org = self.create_organization()
+                assert features.has("organizations:global-views", org)
+
+            def test_method_2(self):
```

Comment:
It may be good to add a test for nested `with_feature` overrides within methods/wrapping methods as well just to ensure we're properly evaluating them.